Sources/SwifterSwift/Foundation/URLExtensions.swift

```diff
@@ -45,9 +45,15 @@ public extension URL {
     /// - Returns: URL with appending given query parameters.
     func appendingQueryParameters(_ parameters: [String: String]) -> URL {
         var urlComponents = URLComponents(url: self, resolvingAgainstBaseURL: true)!
-        var items = urlComponents.queryItems ?? []
-        items += parameters.map({ URLQueryItem(name: $0, value: $1) })
-        urlComponents.queryItems = items
+        var queryItems = urlComponents.queryItems ?? []
+        for param in parameters {
+            if let index = queryItems.firstIndex(where: { $0.name == param.key }) {
+                queryItems[index] = URLQueryItem(name: param.key, value: param.value)
+            } else {
+                queryItems.append(URLQueryItem(name: param.key, value: param.value))
+            }
+        }
+        urlComponents.queryItems = queryItems
```

Comment:
This isn't correct, it's a separate function `settingQueryParameters`. Or maybe you want to add another parameter, `replaceExisting: Bool = false`. But changing this functionality is not correct, query parameters can have duplicate entries.